go-types/parser/run.go

```diff
@@ -0,0 +1,73 @@
+package parser
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"syscall"
+
+	"github.com/estuary/connectors/go-types/firsterror"
```

Comment:
Yea, I went back and forth on this one a bit, and ultimately put them in separate packages just for lack of unifying package name that seemed reasonable. I don't think there's a way to keep the types private, though, since they need to be usable from other packages. Am I missing something there? The alternative would I think be exposing more of a high-level public API like `func RunProgram(...)` that just accepts the callback parameters and a `[]string` for the program/args. But I was hesitant to implement that since it's not clear to me that both the captures runtime and the connectors themselves would actually want the same API.
Ultimately, I'm fine with grouping them into a single package, but I think my preference would be to default to keeping them separate. WDYT?

Reply:
What we're building here is fundamentally a Go development kit for building Airbyte-spec connectors, similar to the python CDK that Airbyte has published. Since this is intended as public facing, the bias must be that we're only making things public when reasonably confident it's the right API -- we should be delaying exporting things as long as possible, because they can't be taken back. For that reason I think the bias is to lump it together in a package as private implementation. I also agree that a `RunProgram` factoring is probably what it will evolve into, but we don't need to figure that out quite yet.